kem/src/lib.rs

```diff
@@ -28,9 +28,15 @@ pub trait Encapsulate<EK, SS> {
 /// Often, this will just be a secret key. But, as with [`Encapsulate`], it can be a bundle
 /// of secret keys, or it can include a sender's private key for authenticated encapsulation.
 pub trait Decapsulate<EK, SS> {
+    /// Encapsulator which corresponds to this decapsulator.
+    type Encapsulator: Encapsulate<EK, SS>;
+
     /// Decapsulation error
     type Error: Debug;
 
     /// Decapsulates the given encapsulated key
     fn decapsulate(&self, encapsulated_key: &EK) -> Result<SS, Self::Error>;
+
+    /// Retrieve the encapsulator associated with this decapsulator.
+    fn to_encapsulator(&self) -> Self::Encapsulator;
```

Comment:
Bikeshed: this could be `encapsulator`?

In the concrete KEM impls there's `encapsulation_key`, but I worry that might get confused with `EK` (i.e. *encapsulated* key)